Assets/Scripts/Ability/AbilityManager.cs

```diff
@@ -0,0 +1,242 @@
+using UnityEngine.InputSystem;
+using System.Collections.Generic;
+using UnityEngine;
+using RogueApeStudio.Crusader.Input;
+using System;
+using System.Threading;
+using UnityEngine.VFX;
+using UnityEditor.Graphs;
+using UnityEditor.Playables;
+
+namespace RogueApeStudio.Crusader.Ability
+{
+    public class AbilityManager : MonoBehaviour
+    {
+        [Header("References")]
+        [SerializeField] private PlayerInput _playerInput;
+        [SerializeField] private Transform _playerCharacterTransform;
+        [SerializeField] private Animator _animator;
+
+        [Header("Cursor Position Necessities")]
+        [SerializeField] private Camera _cam;
+        [SerializeField] private LayerMask _layerMask;
+
+        [Header("Used Abilities")]
+        [SerializeField] private List<Ability> _abilities;
+
+        [Header("Slots")]
+        [SerializeField] private GameObject _slotParent;
+        [SerializeField] private VisualEffect[] _slots;
+
+        private CrusaderInputActions _inputActions;
+        private InputAction _ability1;
+        private InputAction _ability2;
+        private InputAction _ability3;
+        private CancellationTokenSource _cancellationTokenSource;
+        private Dictionary<int, CastTarget> _castTargetCache;
```

Comment:
Why did you go for a `Dictionary `here? Why not a `2D List`, or a `List<CustomStruct>`?

Reply:
This has been resolved verbally.